utils/playbackProfiles/helpers/codecProfiles.ts

```diff
@@ -0,0 +1,345 @@
+import {
+  CodecProfile,
+  CodecType,
+  ProfileConditionType,
+  ProfileCondition,
+  ProfileConditionValue
+} from '@jellyfin/client-axios';
+import { browserDetector } from '~/plugins/browserDetection';
+
+/**
+ * Gets the max video bitrate
+ *
+ * @returns {number | null} Returns the MaxVideoBitrate
+ */
+function getGlobalMaxVideoBitrate(): number | null {
+  let isTizenFhd = false;
+  if (browserDetector.isTizen()) {
+    try {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      // eslint-disable-next-line no-undef
+      const isTizenUhd = webapis?.productinfo?.isUdPanelSupported();
+      isTizenFhd = !isTizenUhd;
+
+      // eslint-disable-next-line no-console
+      console.debug('isTizenFhd = ' + isTizenFhd);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('isUdPanelSupported() error code = ' + error.code);
+    }
+  }
+
+  return browserDetector.isPs4()
+    ? 8000000
+    : browserDetector.isXbox()
+    ? 12000000
+    : browserDetector.isTizen && isTizenFhd
+    ? 20000000
+    : null;
+}
+
+/**
+ * Creates a profile condition object for use in device playback profiles.
+ *
+ * @param {ProfileConditionValue} Property - Value for the property
+ * @param {ProfileConditionType} Condition - Condition that the property must comply with
+ * @param {string} Value - Value to check in the condition
+ * @param {boolean} IsRequired - Whether this property is required
+ * @returns {ProfileCondition} - Constructed ProfileCondition object
+ */
+function createProfileCondition(
+  Property: ProfileConditionValue,
+  Condition: ProfileConditionType,
+  Value: string,
+  IsRequired = false
+): ProfileCondition {
+  return {
+    Condition,
+    Property,
+    Value,
+    IsRequired
+  };
+}
+
+/**
+ * @param {HTMLVideoElement} videoTestElement A HTML video element for testing codecs
+ * @returns {ProfileCondition[]} sd
+ */
+export function getaacCodecProfileConditions(
+  videoTestElement: HTMLVideoElement
+): ProfileCondition[] {
+  const supportsSecondaryAudio = browserDetector.isTizen();
+
+  const conditions = [] as ProfileCondition[];
+
+  // Handle he-aac not supported
+  if (
+    !videoTestElement
+      .canPlayType('video/mp4; codecs="avc1.640029, mp4a.40.5"')
+      .replace(/no/, '')
+  ) {
+    // TODO: This needs to become part of the stream url in order to prevent stream copy
+    conditions.push({
+      Condition: ProfileConditionType.NotEquals,
+      Property: ProfileConditionValue.AudioProfile,
+      Value: 'HE-AAC'
+    });
+  }
+
+  if (!supportsSecondaryAudio) {
+    conditions.push({
+      Condition: ProfileConditionType.Equals,
+      Property: ProfileConditionValue.IsSecondaryAudio,
+      Value: 'false',
+      IsRequired: false
+    });
+  }
+
+  return conditions;
+}
+
+/**
+ *
+ *
+ * @param {HTMLVideoElement} videoTestElement A HTML video element for testing codecs
```

Comment:
```suggestion
 * @param {HTMLVideoElement} videoTestElement - A HTML video element for testing codecs
```

There are more JSDoc in the PR that needs this.

Reply:
Probably easiest to do them all in a single PR. I'll make one after this is merged